What's New 103 (& Experiments update) (#2007)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/whats-new/2022/06/devtools-103.md

See also:
* [Microsoft Edge DevTools extension for Visual Studio Code](../../../../visual-studio-code/microsoft-edge-devtools-extension.md)
<!-- * [Microsoft Edge DevTools for Visual Studio Code](https://aka.ms/devtools-for-code) at Marketplace -->

<!-- ====================================================================== -->
## Bug fixes for Focus Mode UI

<!-- Title: Bug fixes for Focus Mode UI -->
<!-- Subtitle: Various bugs impacting DevTools controls and keyboard shortcuts were fixed in the new Focus Mode UI. -->

In the new **Focus Mode** DevTools UI, various bugs affecting controls and keyboard shortcuts for DevTools have been resolved:
*  Improved behavior of collapsible tabs in the **Elements** tool.
*  Fixed some instances in which the name of an open tool wasn't visible.
*  Improved reliability of **Quick View** controls.
*  Improved behavior of opening DevTools by using keyboard shortcuts.
*  Fixed an issue with navigating to a specific line of code in the **Sources** tool by using keyboard shortcuts.
*  Restored the `Ctrl`+`Shift`+`F` shortcut to open **Search** in **Quick View**:<!-- todo: macOS keys? -->

> 

Reviewer: Mac shortcut is Command+Option+F.

![The Close DevTools (X) button in high contrast mode.](devtools-103-images/close-button-high-contrast.png)

See also:
* [Navigate DevTools with assistive technology](../../../accessibility/navigation.md)
* [Windows high contrast mode](/fluent-ui/web-components/design-system/high-contrast)

<!-- ====================================================================== -->
## Redesigned Welcome tool

<!-- Title: Welcome tool has expanded content and videos -->
<!-- Subtitle: The redesigned Welcome tool has expanded Learn and What's New sections, including a new video series. -->

In Microsoft Edge 103, the **Welcome** tool has a new design with improved, expanded content.  It's easier than ever to find documentation, videos, and information about new features and bug fixes.

*  In the **Learn** section, use the links to quickly jump to documentation for some common tasks in DevTools.

*  In the **What's New** section, find out about the latest changes, new features, and bug fixes.

*  In both sections, learn about existing and new web development features through our new video series, [Videos about web development with Microsoft Edge](../../../../dev-videos/index.md).

> ![The redesigned Welcome tool.](devtools-103-images/welcome-tool.png)

Reviewer 2: Updated screenshot to be inserted here.

In Microsoft Edge 103, when taking a heap snapshot in the **Memory** tool, there are two new node types:
*  **Object shape** - Refers to the hidden classes and descriptor arrays that V8 (the JavaScript engine of Microsoft Edge) uses to understand and index the properties on objects. 
*  **BigInt** - Used to represent and manipulate values that are too large to be represented by the **Number** node type.

![BigInt and Object Shape memory node types.](devtools-103-images/memory-node-types.png)

For history, see [Issue 12769: Improve categorization of objects in heap snapshots](https://bugs.chromium.org/p/v8/issues/detail?id=12769).

See also:
* [Constructor (group) entries in the heap profiler](../../../memory-problems/heap-snapshots.md#constructor-group-entries-in-the-heap-profiler) in _Record heap snapshots using the Memory tool_.
* [HiddenClasses and DescriptorArrays](https://v8.dev/blog/fast-properties#hiddenclasses-and-descriptorarrays) - Fast properties in V8.
* [BigInt object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/BigInt).

<!-- ====================================================================== -->

> ## DevTools extension for Visual Studio Code: Right-click an HTML file to launch, and Application tool

Reviewer 3: I think we can bury the lead a little bit here about the VS Code extension in favor of a shorter, clearer title. The title still has the value prop of being able to do these things directly from VS Code

```suggestion
## Launch the browser from an HTML file and use the Application tool within Visual Studio Code
```

<!-- Title: DevTools for Visual Studio Code: Right-click to launch, and Application tool -->
<!-- Subtitle: Use the context menu on any HTML document in Visual Studio Code to launch a new instance of the Microsoft Edge browser and DevTools. -->

To make it easier to start a new embedded instance of Microsoft Edge from within Visual Studio Code, when you right-click an HTML file in the **Explorer**, there's now an **Open with Edge** submenu that has the commands **Open Browser** and **Open Browser with DevTools**:

![Right-click an HTML file to open a browser instance and the 'Edge DevTools' tab.](devtools-103-images/edge-devtools-rightclick-html-file.png)

> Also, you can now also use the **Application** tool to gain insights into the storage and service worker information of your project, without leaving Visual Studio Code:

Reviewer 4: Repeated "also". Proposal to remove the first one.

![The Application tool in the 'Edge DevTools' tab within Visual Studio Code.](devtools-103-images/edge-devtools-application-tool.png)

See also:
* [Microsoft Edge DevTools extension for Visual Studio Code](../../../../visual-studio-code/microsoft-edge-devtools-extension.md)
<!-- * [Microsoft Edge DevTools for Visual Studio Code](https://aka.ms/devtools-for-code) at Marketplace -->

<!-- ====================================================================== -->
## Bug fixes for Focus Mode UI

<!-- Title: Bug fixes for Focus Mode UI -->
<!-- Subtitle: Various bugs impacting DevTools controls and keyboard shortcuts were fixed in the new Focus Mode UI. -->

In the new **Focus Mode** DevTools UI, various bugs affecting controls and keyboard shortcuts for DevTools have been resolved:

> *  Improved behavior of collapsible tabs in the **Elements** tool.

Reviewer 3: Can we get more specific here? @hxlnt, is this mostly about the tabs in the Styles pane? Or something else in the Elements tool that's collapsible?

Reviewer: Proposed fix:  Improved behavior of collapsible side-panel tabs such as Styles in the **Elements** tool

*  Fixed some instances in which the name of an open tool wasn't visible.
*  Improved reliability of **Quick View** controls.
*  Improved behavior of opening DevTools by using keyboard shortcuts.
*  Fixed an issue with navigating to a specific line of code in the **Sources** tool by using keyboard shortcuts.
*  Restored the `Ctrl`+`Shift`+`F` shortcut to open **Search** in **Quick View**:<!-- todo: macOS keys? -->

![The Search keyboard shortcut opens the Search tool in Quick View.](devtools-103-images/focus-mode-search-shortcut.png)

See also:
* [Simplify DevTools using Focus Mode](../../../experimental-features/focus-mode.md)

<!-- ====================================================================== -->
## The 3D View tool responds better in narrow layouts

<!-- Title: 3D View: Now with more responsive design -->
<!-- Subtitle: The 3D View toolbar now wraps around in a narrower window. -->

> As folks use the **3D View** tool more, especially with the addition of Composited Layers (formerly known as **Layers**), we have found opportunities to improve the UI based on your feedback.  We've made the layout of the 3D View toolbar more responsive so that your development isn't hindered even when DevTools is narrow.  Keep the feedback coming as you check out 3D View!

Reviewer 3: ```suggestion
In previous versions of Microsoft Edge, the **3D View** tool did not display correctly in narrow layouts. The toolbar options were cut off when the DevTools were narrow. In Microsoft Edge 103, this issue has been fixed. The toolbar options in 3D View now stack as the width of the DevTools window is decreased.
```

<!-- Title: Fix: In high contrast mode, the Close button for DevTools is now accessible -->
<!-- Subtitle: In previous versions of Microsoft Edge, the Close button wasn't visible in high contrast mode, but this issue has been fixed in Microsoft Edge 103. -->

In previous versions of Microsoft Edge, the **Close** button for DevTools didn't render correctly in high contrast mode.  In Microsoft Edge 103, this issue has been fixed.  The **Close** (![Close icon.](../../../media/close-devtools-icon-light-theme.png)) button for DevTools is displayed correctly in high contrast mode:

![The Close DevTools (X) button in high contrast mode.](devtools-103-images/close-button-high-contrast.png)

See also:
* [Navigate DevTools with assistive technology](../../../accessibility/navigation.md)
* [Windows high contrast mode](/fluent-ui/web-components/design-system/high-contrast)

<!-- ====================================================================== -->
## Redesigned Welcome tool

<!-- Title: Welcome tool has expanded content and videos -->
<!-- Subtitle: The redesigned Welcome tool has expanded Learn and What's New sections, including a new video series. -->

> In Microsoft Edge 103, the **Welcome** tool has a new design with improved, expanded content.  It's easier than ever to find content and to contact the DevTools team.

Reviewer 3: ```suggestion
In Microsoft Edge 103, the **Welcome** tool has a new design with improved, expanded content.  It's easier than ever to find documentation, videos, and information on new features and bug fixes.
```

We don't have a link to the GitHub repo so I don't think it makes sense to say it's easier to contact the DevTools team. Reeza, we should follow up on adding a GitHub repo link to the Welcome tool

<!-- ![alt text.](devtools-103-images/x.png) -->

<!-- Instructions for screenshot
(how to configure the DevTools correctly, a website or CodePen that the writer can use, where to navigate in the DevTools UI for the screenshot, etc.)
Open Edge Stable and navigate to edge://version and ensure you're on version 102 or at least <103
From Windows Settings, navigate to Accessibility > Contrast themes and select the Night sky theme to apply
Open DevTools in Edge Stable. Note that the Close button won't be visible (you can still select it though).
Refer to the attachment for reference
Repeat the same steps in Edge Canary or Dev, just ensure the version is 103+. Refer to the attachment for reference. -->

<!-- Video recording of feature in action
.mp4 attachments -->

See also:
* [Navigate DevTools with assistive technology](../../../accessibility/navigation.md)
* [Windows High contrast mode](/fluent-ui/web-components/design-system/high-contrast)

<!-- ====================================================================== -->

> ## Welcome tool redesign

Author: @Reezaali - ready for title/subtitle/ draft text.  rendered:
https://review.docs.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/whats-new/2022/06/devtools-103?branch=pr-en-us-2007#welcome-tool-redesign
